virttest/qemu_monitor.py

```diff
@@ -629,6 +631,50 @@ def wait_for_migrate_progress(self, target):
             old_progress = progress
             time.sleep(0.1)
 
+    def _get_migrate_capability(self, capability):
+        """
+        Verify the $capability is listed in migrate-capabilitys. If not try
+        x-/non-x- version. In case none is supported, return the original param
```

Comment:
Suggest making the behavior configurable: by enabled, automatically chose supported options; by disabled, uses user specified ones.

Reply:
Would it be fine to leave it as is and wait for someone actively requiring this?

Reply:
It is acceptable to me that make the `x` to `non-x` conversion without using a param as the flag, but I think it's not good to do so for the counter one because verifying the existence of formal capabilities/parameters is one of the checkpoints, and it should be a positive test.

Reply:
Do you have an example of such test? I mean do we want it by default to be disabled, or enabled? Do we want the `vm.migrate` to allow enabling/disabling it (therefore yet another argument)?

Reply:
Let me summarize my perspective: let tests do what they exactly want to do, not the modified version provided by the vt framework, unless it is what they required.

So,
> Do you have an example of such test?

Let's take a look at the test which is supposed to be benefited,
```
# qemu/tests/cfg/migrate.cfg
...
                - with_multifd:
                    only tcp,unix,fd
                    migrate_capabilities = "{'x-multifd': 'on'}"
                    migrate_parameters = "{'x-multifd-channels': 8, 'x-multifd-page-count': 64}"
                    target_migrate_parameters = "{'x-multifd-channels': 8, 'x-multifd-page-count': 64}"
...
```
at least I can't say it doesn't care about checking the existence of capabilities/parameters because the author left nothing but codes. So as a result the feature introduced by this PR would redefine the usage of the above three at the bottom, that's ok, but please leave a comment or write a doc to clarify that, will you? not only for this one, but also for the rest.

> I mean do we want it by default to be disabled, or enabled?

To my perspective, I'd suggest making it be disabled as the default, but if you think the majority of tests do require this feature you can make it be enabled. With the param, for people who don't want it, like me, the feature can be disabled easily with using their own customized configurations.

> Do we want the vm.migrate to allow enabling/disabling it (therefore yet another argument)?

Introducing another argument to the `vm.migrate()` is just one of the approaches, you can do it by introducing another attribute to the `VM` class as a flag (e.g. `vm.auto_x_feature_conversion`) and set it according to user's configuration in `__init__()`, and then reference it in `migrate()`. Please be noted that the `x-` things handled in this PR are not only the ones to be benefited, there are other experimental features, for example the things are handled in https://github.com/avocado-framework/avocado-vt/pull/1932, so just don't worry about it is an "unnecessary" flag.

Reply:
Well, I guess this feature is closer to selecting the right machine type, cpu_model, ide vs. ahci or such. Anyway I understand you insist on a way to disable it, so I have added the e577927.

I kept the auto-selecting on by default as all migration tests would have to be adjusted as we want this.

On the other hand when directly using qemu monitor I disabled it by default, therefor low-level interface should be clean and high level interface should be benevolent.

Does it make sense like this?